homeassistant/components/homeassistant/exposed_entities.py

```diff
@@ -0,0 +1,302 @@
+        self._store: Store[dict[str, dict[str, dict[str, Any]]]] = Store(
+            hass, STORAGE_VERSION, STORAGE_KEY
+        )
+
+    async def async_initialize(self) -> None:
+        """Finish initializing."""
+        websocket_api.async_register_command(self._hass, ws_expose_entity)
+        websocket_api.async_register_command(self._hass, ws_expose_new_entities)
+        await self.async_load()
+
+    @callback
+    def async_listen_entity_updates(
+        self, assistant: str, listener: Callable[[], None]
+    ) -> None:
+        """Listen for updates to entity expose settings."""
+        self._listeners.setdefault(assistant, []).append(listener)
+
+    @callback
+    def async_expose_entity(
+        self, assistant: str, entity_id: str, should_expose: bool
+    ) -> None:
+        """Expose an entity to an assistant.
+
+        Notify listeners if expose flag was changed.
+        """
+        entity_registry = er.async_get(self._hass)
+        if not (registry_entry := entity_registry.async_get(entity_id)):
+            return
+
+        if (
+            assistant_options := registry_entry.options.get(assistant)
+        ) and assistant_options["should_expose"] == should_expose:
+            return
+
+        assistant_options = {"should_expose": should_expose}
+        entity_registry.async_update_entity_options(
+            entity_id, assistant, assistant_options
+        )
+        for listener in self._listeners.get(assistant, []):
+            listener()
+
+    @callback
+    def async_expose_new_entities(self, assistant: str, expose_new: bool) -> None:
+        """Enable an assistant to expose new entities."""
+
+        self._assistants[assistant] = AssistantPreferences(expose_new=expose_new)
+        self._async_schedule_save()
+
+    @callback
+    def async_get_exposed_entities(
+        self, assistant: str
+    ) -> dict[str, Mapping[str, Any]]:
+        """Get all exposed entities."""
+        entity_registry = er.async_get(self._hass)
+        result = {}
+
+        for entity_id, entry in entity_registry.entities.items():
+            if options := entry.options.get(assistant):
+                result[entity_id] = options
+
+        return result
+
+    @callback
+    def async_should_expose(self, assistant: str, entity_id: str) -> bool:
+        """Return True if an entity should be exposed to an assistant."""
+        should_expose: bool
+
+        if entity_id in CLOUD_NEVER_EXPOSED_ENTITIES:
+            return False
+
+        entity_registry = er.async_get(self._hass)
+        if not (registry_entry := entity_registry.async_get(entity_id)):
+            # Entities which are not in the entity registry are not exposed
+            return False
+
+        if assistant in registry_entry.options:
+            should_expose = registry_entry.options[assistant]["should_expose"]
+            return should_expose
+
+        if not (prefs := self._assistants.get(assistant)) or not prefs.expose_new:
```

Comment:
since there is an else-statement, you can remove the `not's` here and make it a bit more readable.